src/Stormpath/Client.php

```diff
@@ -89,30 +90,31 @@ class Client extends Magic
 
     private static $instance;
 
-    private $cacheManagerInstance;
-
     private $dataStore;
 
     /**
      * Instantiates a new Client instance that will communicate with the Stormpath REST API.
      * See the class-level PHPDoc for a usage example.
      *
-     * @param $apiKey the Stormpath account API Key that will be used to authenticate the client with
+     * @param ApiKey| $apiKey the Stormpath account API Key that will be used to authenticate the client with
      *               Stormpath's REST API.
      *
-     * @param $baseUrl optional parameter for specifying the base URL when not using the default one
+     * @param CacheManager $cacheManager
+     * @param $baseUrl parameter for specifying the base URL when not using the default one
      *         (https://api.stormpath.com/v1).
+     * @param RequestSigner $requestSigner
      */
-    public function __construct(ApiKey $apiKey, $cacheManager, $cacheManagerOptions, $baseUrl = null, RequestSigner $requestSigner = null)
+    public function __construct(ApiKey $apiKey, CacheManager $cacheManager = null, $baseUrl = null, RequestSigner $requestSigner = null)
```

Comment:
This will break people's code if they are creating the client using the constructor directly. We might need to wait for the next major release version to change this.